editor/components/block-switcher/index.js

```diff
@@ -17,20 +18,26 @@ import { keycodes } from '@wordpress/utils';
  */
 import './style.scss';
 import { replaceBlocks } from '../../store/actions';
-import { getBlock } from '../../store/selectors';
+import { getBlock, getCurrentPostType } from '../../store/selectors';
 
 /**
  * Module Constants
  */
 const { DOWN } = keycodes;
 
-function BlockSwitcher( { blocks, onTransform, isLocked } ) {
+function BlockSwitcher( { blocks, onTransform, isLocked, user } ) {
 	const allowedBlocks = getPossibleBlockTransformations( blocks );
+	const userCapabilities = get( user.data, 'capabilities' );
 
 	if ( isLocked || ! allowedBlocks.length ) {
 		return null;
 	}
 
+	const userAllowedBlocks = allowedBlocks.filter( b => ! b.capability || userCapabilities[ b.capability ] );
```

Comment:
I think there's a chance `userCapabilities` could be undefined here; likely not observed in general usage only because we [preload the `/users/me` request](https://github.com/WordPress/gutenberg/blob/75d81e9d13c065fc80d3184fe9ac58c0da0f0573/lib/client-assets.php#L792) but something which could easily break in future refactoring.

One option is to provide the [third argument to Lodash's `_.get`](https://lodash.com/docs/4.17.4#get), which is a default value in case `capabilities` is not available. We could assign this to an empty object.

Or: We could bypass the filter altogether by wrapping this in an `if` condition or including `userCapabilities` in the above condition to `return null;`.

Aside: I have a personal vendetta against abbreviated variable names. I'd rather we assign a full variable name `block` and just split this into multiple lines. The single line doesn't necessarily help readability, or at least not as much as I think the abbreviated variable name harms it.